src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Syntax/SyntaxWalker.cs

```diff
@@ -18,35 +16,48 @@ internal abstract class SyntaxWalker : SyntaxVisitor
     private int _recursionDepth;
     private readonly TextSpan? _range;
 
-    public SyntaxWalker(TextSpan? range = null)
+    protected SyntaxWalker(TextSpan? range = null)
     {
         _range = range;
     }
 
-    public override void Visit(SyntaxNode node)
+    private bool ShouldVisit(TextSpan span)
+    {
+        return _range is TextSpan range && range.OverlapsWith(span);
```

Comment:
Is this correct? Previously we would visit if `_range` was `null`, but now we won't?

Reply:
Good catch! However, I ended up removing the range check from `SyntaxWalker` in a later commit (7ce5005a7e8b15255c20d443390a8aadfe31f0cc). This check never really belonged here and was only used by one `SyntaxWalker` descendant: `SemanticTokensVisitor`. It turns out that I left the seem logic error there as well, but `_range` is never passed null. So, it works, but I'll clean it up.